### PR TITLE
Adds new metadata values

### PR DESCRIPTION
**Pull request by chentex**
## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Documentation Update

## Description

- Includes FIPS, Publish, and Architecture extraction

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please describe the System Under Test.
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

**Comment by jtaleric** [comment]
Curious, what is Publish?

**Comment by chentex** [comment]
> Curious, what is Publish?

It tells you if the cluster is External or Private facing, I ma open to new names

File: ocp-metadata/types.go [hunk 1]
```diff
@@ -86,4 +86,8 @@ type ClusterMetadata struct {
 	SDNType          string `json:"sdnType"`
 	ClusterName      string `json:"clusterName"`
 	Region           string `json:"region"`
+	Fips             bool   `json:"fips"`
+	Publish          string `json:"publish"`
+	WorkerArch       string `json:"wrkerArch"`
```
Comment (rsevilla87) [inline ocp-metadata/types.go:91]: typo?

**Review by rsevilla87 — COMMENTED** [review]
Remove d

**Comment by vishnuchalla** [comment]
> > Curious, what is Publish?
> 
> It tells you if the cluster is External or Private facing, I'm open to new names

if its just a bool, can we rename it as `isPublic` or something?

File: ocp-metadata/ocp-metadata.go [hunk 2]
```diff
@@ -270,3 +293,110 @@ func (meta *Metadata) getSDNInfo() (string, error) {
 	}
 	return networkType, err
 }
+
+func (meta *Metadata) getPublish() (string, error) {
+	installConfig, err := meta.getClusterConfig()
```
Comment (rsevilla87) [inline ocp-metadata/ocp-metadata.go:298]: You're calling getClusterConfig many times along the different methods, I wonder if we can call this function only once

**Review by rsevilla87 — CHANGES_REQUESTED** [review]
The current metadata implementation doesn't crash when used on vanilla kubernetes clusters (kind), I wonder if it's possible to follow this strategy with the new fields.

For example, the library was handling the error of the infra object in these lines:


https://github.com/cloud-bulldozer/go-commons/blob/0140b108fb4b1723c1a94330acfdd7392efa2a7e/ocp-metadata/ocp-metadata.go#L173-L178

**Comment by chentex** [comment]
Resolved all the comments.

**Review by rsevilla87 — APPROVED** [review]
lgtm,
works in OCP and Vanillla k8s

Vanilla k8s
```shell
$ go run main.go  | jq .
{
  "k8sVersion": "v1.29.2",
  "masterNodesCount": 1,
  "totalNodes": 1
}
```

OCP 4 .12
```shell
$ go run main.go  | jq .
{
  "platform": "AWS",
  "clusterType": "self-managed",
  "ocpVersion": "4.14.3",
  "ocpMajorVersion": "4.14",
  "k8sVersion": "v1.27.6+b49f9d1",
  "masterNodesType": "m6i.2xlarge",
  "masterNodesCount": 3,
  "workerNodesCount": 3,
  "totalNodes": 3,
  "sdnType": "OVNKubernetes",
  "clusterName": "ocp-ci-hq68v",
  "region": "us-west-2",
  "publish": "External",
  "workerArch": "amd64",
  "controlPlaneArch": "amd64",
  "ipsecMode": "Disabled"
}

```

**Review by vishnuchalla — APPROVED** [review]

**Pull request merged**